### ISSUE TITLE
Remove usage of trace package

### ISSUE THREAD

**Issue by jeroenrinzema** [jexia/semaphore#137]
The [trace](https://github.com/jexia/semaphore/tree/master/pkg/broker/trace) package has been used to construct error messages for unexpected behavior. The `trace` package is to be replaced with error wrapping and custom [error types](https://github.com/jexia/semaphore/pull/128/files). Any last references to this package could be find by [searching for](https://github.com/jexia/semaphore/search?q=trace.New) `trace.New` inside the project.

Custom errors could implement the [pretty interface](https://github.com/jexia/semaphore/blob/master/pkg/prettyerr/prettifier.go#L3-L7) to describe detailed information about the given error.

```go
// ErrUndefinedFunction occurs when custom function is not defined in property but referenced
type ErrUndefinedFunction struct {
	Function string
	Property string
}

// Error returns a description of the given error as a string
func (e ErrUndefinedFunction) Error() string {
	return fmt.Sprintf("undefined custom function '%s' in '%s'", e.Function, e.Property)
}

// Prettify returns the prettified version of the given error
func (e ErrUndefinedFunction) Prettify() prettyerr.Error {
	return prettyerr.Error{
		Code:    "UndefinedFunction",
		Message: e.Error(),
		Details: map[string]interface{}{
			"Function": e.Function,
			"Property": e.Property,
		},
	}
}
```

**Comment by ahmedkrmn** [comment]
Hi @jeroenrinzema,
Can I work on this one?

**Comment by jeroenrinzema** [comment]
Hi @ahmedkrmn, sounds great! Feel free to open a PR once you are done.

**Comment by veger** [comment]
Implemented by #166
